prometheus/registry.go

```diff
@@ -252,9 +252,9 @@ func (errs MultiError) MaybeUnwrap() error {
 }
 
 // Registry registers Prometheus collectors, collects their metrics, and gathers
-// them into MetricFamilies for exposition. It implements both Registerer and
-// Gatherer. The zero value is not usable. Create instances with NewRegistry or
-// NewPedanticRegistry.
+// them into MetricFamilies for exposition. It implements Registerer, Gatherer,
+// and Collector. The zero value is not usable. Create instances with
+// NewRegistry or NewPedanticRegistry.
```

Comment:
Do you mind adding a paraph about implementing Collector, it's behavior and what it enables?

Ideally we could add short `ExampleRegistryGrouping`  example in test file too! 🤗